### PR TITLE
libnotify: 0.7.7 -> 0.7.8

### PR DESCRIPTION
**Pull request by dtzWill**
* cleanup a bit using new helpers
* move to meson to avoid need for gtk_doc and such
* NEWS: https://gitlab.gnome.org/GNOME/libnotify/blob/0.7.8/NEWS
* tag: https://gitlab.gnome.org/GNOME/libnotify/tags/0.7.8

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by xeji** [comment]
darwin failure is a regression and should be fixed before merging.

**Comment by flokli** [comment]
while switching to meson, libnotify/meson.build started setting the `--version-script` parameter, which doesn't exist on the MacOS linker.

We could patch it out, or upstream some sort of detection, similar to here: 
https://github.com/mkhl/mrsh/commit/6a31bd9754fcf2ef9579eea65d271341ba1610e3#diff-969b60ad3d206fd45c208e266ccfed38R43

File: pkgs/development/libraries/libnotify/default.nix [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
   src = fetchurl {
-    url = "mirror://gnome/sources/libnotify/${ver_maj}/${name}.tar.xz";
-    sha256 = "017wgq9n00hx39n0hm784zn18hl721hbaijda868cm96bcqwxd4w";
+    url = "mirror://gnome/sources/libnotify/${stdenv.lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
+    sha256 = "1371csx0n92g60b5dmai4mmzdnx8081mc3kcgc6a0xipcq5rw839";
   };
 
-  # disable tests as we don't need to depend on gtk+(2/3)
-  configureFlags = [ "--disable-tests" ];
+  mesonFlags = [ "-Dtests=false" "-Ddocbook_docs=disabled" "-Dgtk_doc=false" ];
```
Comment (jtojnar) [inline pkgs/development/libraries/libnotify/default.nix]: For cleaner diff in the future

```suggestion
  mesonFlags = [
    "-Dtests=false"
    "-Ddocbook_docs=disabled"
    "-Dgtk_doc=false"
  ];
```

File: pkgs/development/libraries/libnotify/default.nix [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
-  ver_maj = "0.7";
-  ver_min = "7";
-  name = "libnotify-${ver_maj}.${ver_min}";
+  pname = "libnotify";
+  version = "0.7.8";
 
   src = fetchurl {
-    url = "mirror://gnome/sources/libnotify/${ver_maj}/${name}.tar.xz";
-    sha256 = "017wgq9n00hx39n0hm784zn18hl721hbaijda868cm96bcqwxd4w";
+    url = "mirror://gnome/sources/libnotify/${stdenv.lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
```
Comment (jtojnar) [inline pkgs/development/libraries/libnotify/default.nix]: Might as well:

```suggestion
    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
```

File: pkgs/development/libraries/libnotify/default.nix [hunk 3]
```diff
@@ -1,20 +1,18 @@
-{ stdenv, fetchurl, pkgconfig, autoreconfHook
+{ stdenv, fetchurl, meson, ninja, pkgconfig
 , glib, gdk_pixbuf, gobject-introspection }:
```
Comment (jtojnar) [inline pkgs/development/libraries/libnotify/default.nix]:
```suggestion
, glib, gdk_pixbuf, gobject-introspection, gnome3 }:
```

File: pkgs/development/libraries/libnotify/default.nix [hunk 4]
```diff
@@ -1,20 +1,18 @@
...
+    sha256 = "1371csx0n92g60b5dmai4mmzdnx8081mc3kcgc6a0xipcq5rw839";
   };
 
-  # disable tests as we don't need to depend on gtk+(2/3)
-  configureFlags = [ "--disable-tests" ];
+  mesonFlags = [ "-Dtests=false" "-Ddocbook_docs=disabled" "-Dgtk_doc=false" ];
 
-  nativeBuildInputs = [ pkgconfig autoreconfHook gobject-introspection ];
+  nativeBuildInputs = [ meson ninja pkgconfig gobject-introspection ];
   buildInputs = [ glib gdk_pixbuf ];
```
Comment (jtojnar) [inline pkgs/development/libraries/libnotify/default.nix]: As per https://github.com/NixOS/nixpkgs/issues/36150:

```suggestion

  passthru = {
    updateScript = gnome3.updateScript {
      packageName = pname;
      versionPolicy = "none";
    };
  };

```

**Comment by jtojnar** [comment]
Here is an upstream MR for the Darwin failure: https://gitlab.gnome.org/GNOME/libnotify/merge_requests/9

File: pkgs/development/libraries/libnotify/default.nix [hunk 5]
```diff
@@ -1,22 +1,34 @@
...
-  configureFlags = [ "--disable-tests" ];
+  mesonFlags = [
+    # disable tests as we don't need to depend on gtk+(2/3)
+    "-Dtests=false"
+    "-Ddocbook_docs=disabled"
+    "-Dgtk_doc=false"
+  ];
 
-  nativeBuildInputs = [ pkgconfig autoreconfHook gobject-introspection ];
+  patches = stdenv.lib.optional stdenv.hostPlatform.isDarwin ./no-version-script.patch;
```
Comment (flokli) [inline pkgs/development/libraries/libnotify/default.nix]: can we fetchpatch from https://gitlab.gnome.org/GNOME/libnotify/commit/55eb69247fe2b479ea43311503042fc03bf4e67d.patch (with a comment linking to https://gitlab.gnome.org/GNOME/libnotify/merge_requests/9) instead?

**Review by flokli — DISMISSED** [review]

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build libnotify

**Comment by worldofpeace** [comment]
Thanks everyone :sparkles:

**Pull request merged**